src/Connection/Http2Connection.php

```diff
@@ -75,6 +75,9 @@ final class Http2Connection implements Connection
     public const INADEQUATE_SECURITY = 0xc;
     public const HTTP_1_1_REQUIRED = 0xd;
 
+    public const DEFAULT_MAX_HEADER_SIZE = 1 << 20;
```

Comment:
1MB for headers is a bit much, no?

Reply:
Probably. We'll have to look at if and how we want these settings to be configured and pick some defaults.

Reply:
Should not matter too much as a client. TBH, if you do random requests to not-somewhat-known sites, then you need sanity checks at every layer.